nav2_system_tests/src/system/test_system_node.py

```diff
@@ -152,7 +180,11 @@ def test_all(test_robot):
 
 def main(argv=sys.argv[1:]):
     rclpy.init()
-
+    global random_test_flag
```

Comment:
These shouldn't need to be global variables, they should be passed as parameters to test_all. Also I would change the names to random_goal and num_retries to be more clear